### PR TITLE
Local client and UI tweaks and bunch of other stuff

### PR DESCRIPTION
**Pull request by Nepomuceno**
### All Submissions:

* [ ] Have you followed the guidelines in our Contributing document?
* [ ] Have you successfully ran tests with your changes locally? Including container validation.
* [ ] Have you lint your code locally prior to submission?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ ] Have you labelled this PR correctly? E.G. major, milestone, breaking, breaking-change, minor, feature, enhancement, patch, fix, chore, refactor etc

### Changes to Core Features:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?

### CODE REVIEW

File: client/src/Components/Annotation/AnnotationViewer.tsx [hunk 1]
```diff
@@ -25,24 +36,51 @@ const AnnotationViewer = (props) => {
...
+      (annotations[annot_indx].y2 - annotations[annot_indx].y1) * fftSize * zoomLevel;
+    meta.annotations[f]['core:freq_lower_edge'] =
       (annotations[annot_indx].x1 / fftSize) * meta.global['core:sample_rate'] + lower_freq;
-    updatedAnnotation['core:freq_upper_edge'] =
+    meta.annotations[f]['core:freq_upper_edge'] =
       (annotations[annot_indx].x2 / fftSize) * meta.global['core:sample_rate'] + lower_freq;
-    dispatch(setMetaAnnotation({ annotation: updatedAnnotation, index: f }));
+    updateMeta(queryClient, meta);
+
+    // Now update the actual meta.annotations
```
Comment (sebastus) [inline client/src/Components/Annotation/AnnotationViewer.tsx:51]: Is this commented code needed for something?

File: client/src/Components/Annotation/AnnotationViewer.tsx [hunk 2]
```diff
@@ -25,24 +36,51 @@ const AnnotationViewer = (props) => {
...
-    const f = annotations[annot_indx]['index']; // remember there are 2 different indexes- the ones on the screen and the meta.annotations
-    const updatedAnnotations = [...meta.annotations];
-    const updatedAnnotation = { ...updatedAnnotations[f] };
     const start_sample_index = lowerTile * TILE_SIZE_IN_IQ_SAMPLES;
-    updatedAnnotation['core:sample_start'] = annotations[annot_indx].y1 * fftSize * zoomLevel + start_sample_index;
-    updatedAnnotation['core:sample_count'] =
-      (annotations[annot_indx].y2 - annotations[annot_indx].y1) * fftSize * zoomLevel;
     const lower_freq = meta.captures[0]['core:frequency'] - meta.global['core:sample_rate'] / 2;
-    updatedAnnotation['core:freq_lower_edge'] =
+    const f = annotations[annot_indx]['index'];
```
Comment (sebastus) [inline client/src/Components/Annotation/AnnotationViewer.tsx:41]: Would it make sense to encapsulate some of these calculations as well? Such as setSampleStart(...), setSampleCount(...) and so on.

**Review by sebastus — COMMENTED** [review]
I left a few comments. Honestly, with this much change, I say we test, commit, then fix what's broken as we go. Impossible to pick up anything meaningful with this much change.

File: client/src/Components/Annotation/AnnotationList.tsx [hunk 3]
```diff
@@ -29,32 +32,35 @@ export const AnnotationList = ({ updateSpectrogram }) => {
...
+    [meta, spectrogramHeight, setHandleTop]
   );
 
   const updateAnnotation = useCallback(
     (value, parent) => {
+      if (!meta) return;
       let newAnnotationValue = value;
 
       // Get the min and max frequencies
-      const minFreq = meta.captures[0]['core:frequency'] - meta.global['core:sample_rate'] / 2;
```
[the remark this answers is not in the document]
Comment (Nepomuceno) [reply client/src/Components/Annotation/AnnotationList.tsx:48]: Great idea to do the that in the metadtada file.

File: client/src/Components/Annotation/AnnotationViewer.tsx (hunk 1 above)
Comment (Nepomuceno) [reply client/src/Components/Annotation/AnnotationViewer.tsx:51]: THis is the code to do the update in the annotsations this need to be back before we can merge the PR without this the annostations will not work

File: client/src/Components/Annotation/AnnotationViewer.tsx [hunk 4]
```diff
@@ -59,26 +97,37 @@ const AnnotationViewer = (props) => {
...
+      Object.assign(new Annotation(), {
+        'core:sample_start': annotations[annot_indx].y1 * fftSize * zoomLevel + start_sample_index,
+        'core:sample_count': (annotations[annot_indx].y2 - annotations[annot_indx].y1) * fftSize * zoomLevel,
+        'core:freq_lower_edge': (annotations[annot_indx].x1 / fftSize) * meta.global['core:sample_rate'] + lower_freq,
+        'core:freq_upper_edge': (annotations[annot_indx].x2 / fftSize) * meta.global['core:sample_rate'] + lower_freq,
+        'core:description': annotations[annot_indx]['description'],
+      })
+    );
+    updateMeta(queryClient, meta);
+    // const f = updatedAnnotations.length - 1;
```
[the remark this answers is not in the document]
Comment (Nepomuceno) [reply client/src/Components/Annotation/AnnotationViewer.tsx:120]: Sam thing as the other code this is the code that actually saves the annotations.

**Review by Jaydee123 — APPROVED** [review]

File: client/src/Components/Annotation/AnnotationViewer.tsx (hunk 2 above)
Comment (Nepomuceno) [reply client/src/Components/Annotation/AnnotationViewer.tsx:41]: Probably something we could add to new prs

**Pull request merged**